cirq-core/cirq/sim/clifford/clifford_simulator_test.py

```diff
@@ -546,7 +546,7 @@ def test_valid_apply_measurement():
     state = cirq.CliffordState(qubit_map={q0: 0}, initial_state=1)
     measurements = {}
     _ = state.apply_measurement(cirq.measure(q0), measurements, np.random.RandomState())
-    assert measurements == {'0': [1]}
+    assert measurements == {'0': (1,)}
```

Comment:
While the behavior of `[1]` and `(1,)` are mostly the same, this feels like a red flag to me. Why does the type of `measurements` change, and how does this affect other parts of Cirq?

Reply:
Yeah, this bugs me too. In `ClassicalData` I store measurements as tuples, just because immutability. Mostly measurements are accessed via `log_of_measurement_results` so I change those tuples to lists there for backwards compatibility. This is the one other place that exposes measurements and I couldn't initially figure out a way to make this one backwards compatible, and it doesn't seem like a function that would be used by anyone outside of cirq itself. I'll try again to see if there's a way to make this backwards compatible too though. Otherwise we can either revert back to List for measurements, or accept the minor breaking change. (Sounds like immutability isn't a big thing in the Python ecosystem anyway).